mountpoint-s3/src/inode.rs

```diff
@@ -242,11 +242,14 @@ impl Superblock {
             return Err(InodeError::SetAttrNotPermittedOnRemoteInode(inode.err()));
         }
 
-        // Should be impossible since local file stat never expire.
-        if !sync.stat.is_valid() {
-            error!(?ino, "local inode stat already expired");
-            return Err(InodeError::SetAttrOnExpiredStat(inode.err()));
-        }
+        let validiy = match inode.kind() {
+            InodeKind::File => self.inner.cache_config.file_ttl,
+            InodeKind::Directory => self.inner.cache_config.dir_ttl,
+        };
+
+        sync.stat.expiry = Instant::now()
```

Comment:
Why are we refreshing the validity here? If required, shouldn't we do it in `update_slow_path`?

Reply:
Also, why not call `update_validity`?

Reply:
Yes, I should use `update_validity`. We did not do this `update_slow_path` because it does not update the Inode Stat for local inode. This case is specifically for Local Inode as we are setting new InodeStat, we want it to have renewed expiry.

Reply:
That's my point: now that we are not using NEVER_EXPIRES_TTL anymore for local inodes, maybe we should update their stats on lookup, likely in `update_slow_path`. Otherwise, local inodes will never have their initial expiry updated, except on calls to `settattr` (why?), and will always immediately trigger an S3 lookup to confirm they are still local.